string topColour
string bottomColour
float32 x
float32 y
